packages/core/src/agent/TransportService.ts

```diff
@@ -40,6 +48,12 @@ export class TransportService {
   public removeSession(session: TransportSession) {
     delete this.transportSessionTable[session.id]
   }
+
+  private getWebSocketSessionsForConnectionId(connectionId: string) {
+    return Object.values(this.transportSessionTable).filter(
+      (session) => session?.connectionId === connectionId && session.type === 'WebSocket'
+    )
+  }
 }
```

Comment:
Maybe we can limit it to one session per type and connection id? That way we don't have to add web socket specific logic?

Reply:
Yeah, I'm happy with that also. Bug was only found in WebSocket, that's why I made it this way, but happy to change :)